### PR TITLE
Mention recursion as an attack vector

### PR DESCRIPTION
**Pull request by glyn**
Also add some general advice from https://www.rfc-editor.org/rfc/rfc8949#name-security-considerations.

See also [this rendered version](https://ietf-wg-jsonpath.github.io/draft-ietf-jsonpath-base/186-recursion-DOS/draft-ietf-jsonpath-base.html#name-attack-vectors-on-jsonpath-).

Ref: https://github.com/ietf-wg-jsonpath/draft-ietf-jsonpath-base/issues/186#issuecomment-1158920404

### CODE REVIEW

**Review by timbray — APPROVED** [review]

**Review by cabo — CHANGES_REQUESTED** [review]

File: draft-ietf-jsonpath-base.md [hunk 1]
```diff
@@ -1504,8 +1504,10 @@ and {{Section 8 of -iregexp}} for performance issues in regular
 expression implementations.)
 Implementers need to be aware that good average performance is not
 sufficient as long as an attacker can choose to submit specially
-crafted JSONPath queries that trigger surprisingly high, possibly
-exponential, CPU usage.
+crafted JSONPath queries or arguments that trigger surprisingly high, possibly
+exponential, CPU usage or, for example via the descendant selector,
```
[the remark this answers is not in the document]
Comment (glyn) [reply draft-ietf-jsonpath-base.md:1508]: That example points out the main source of recursion in JSONPath implementations and will hopefully make implementers think twice about how they implement this feature.

(The other potential source of recursion would be nested filters, but our grammar currently precludes these.)

[the remark this answers is not in the document]
Comment (timbray) [reply draft-ietf-jsonpath-base.md:1508]: I think the descendent-selector mention is OK, probably adds value.  But maybe find a way to use the word "recursion"?

**Comment by cabo** [comment]
On 24. Jun 2022, at 18:03, Tim Bray ***@***.***> wrote:
> 
> But maybe find a way to use the word "recursion”?

Recursion is not really the problem, stack depth is (or any other uncontrolled resource usage triggered by queries into deeply nested structures).

Grüße, Carsten

**Comment by glyn** [comment]
@cabo Are you comfortable with this now, or would you perhaps like to suggest some other way to address the concerns of https://github.com/ietf-wg-jsonpath/draft-ietf-jsonpath-base/issues/186?

File: draft-ietf-jsonpath-base.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (cabo) [reply draft-ietf-jsonpath-base.md:1508]:
```suggestion
exponential, CPU usage or, for example via a naive recursive implementation for the descendant selector,
```

[the remark this answers is not in the document]
Comment (glyn) [reply draft-ietf-jsonpath-base.md]: @cabo Applied the change, thanks. Please could you approve the PR?

**Review by cabo — APPROVED** [review]
Thank you.

**Pull request merged**